include/wlr/types/wlr_tablet_v2.h

```diff
@@ -0,0 +1,172 @@
+#ifndef WLR_TYPES_WLR_TABLET_V2_H
+#define WLR_TYPES_WLR_TABLET_V2_H
+
+#include <wayland-server.h>
+#include <wlr/types/wlr_seat.h>
+#include <wlr/types/wlr_input_device.h>
+
+#include "tablet-unstable-v2-protocol.h"
+
+/* This can probably be even lower,the tools don't have a lot of buttons */
+#define WLR_TABLET_V2_TOOL_BUTTONS_CAP 16
+
+struct wlr_tablet_client_v2;
+struct wlr_tablet_tool_client_v2;
+struct wlr_tablet_pad_client_v2;
+
+struct wlr_tablet_manager_v2 {
+	struct wl_global *wl_global;
+	struct wl_list clients; // wlr_tablet_manager_client_v2::link
+	struct wl_list seats; // wlr_tablet_seat_v2::link
+
+	struct wl_listener display_destroy;
+
+	void *data;
+};
+
+struct wlr_tablet_v2_tablet {
+	struct wl_list link; // wlr_tablet_seat_v2::tablets
+	struct wlr_tablet_tool *wlr_tool;
+	struct wlr_input_device *wlr_device;
+	struct wl_list clients; // wlr_tablet_client_v2::tablet_link
+
+	struct wl_listener tool_destroy;
+
+	struct wlr_tablet_client_v2 *current_client;
+};
+
+struct wlr_tablet_v2_tablet_tool {
+	struct wl_list link; // wlr_tablet_seat_v2::tablets
+	struct wlr_tablet_tool_tool *wlr_tool;
+	struct wl_list clients; // wlr_tablet_tool_client_v2::tablet_link
+
+	struct wl_listener tool_destroy;
+
+	struct wlr_tablet_tool_client_v2 *current_client;
+	struct wlr_surface *focused_surface;
+	struct wl_listener surface_destroy;
+	struct wl_listener client_destroy;
```

Comment:
There are a lot of different clients in here. I think this should be renamed to `seat_client_destroy` for clarity.